File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int ,int> mp;
        vector<int> temp;
        for(int i : arr) mp[i]++;

        for(auto it : mp){
            temp.push_back(it.second);
        }
        sort(temp.begin() , temp.end());

        for(int i = 1; i < temp.size(); i++){
            if(temp[i-1] == temp[i])
            return false;
        }
        return true;
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution(object):
    def uniqueOccurrences(self, arr):

        mp = {}
        temp = []

        for i in range (len(arr)):
            mp[arr[i]] = mp.get(arr[i] , 0) + 1
        
        for i in mp.values():
            temp.append(i)

        temp.sort()

        for i in range (len(temp) - 1):
            if temp[i] == temp[i + 1]:
                return False

        return True        